presto-spi/src/main/java/com/facebook/presto/spi/type/TimeType.java

```diff
@@ -39,7 +39,12 @@ public Object getObjectValue(ConnectorSession session, Block block, int position
             return null;
         }
 
-        return new SqlTime(block.getLong(position, 0), session.getTimeZoneKey());
+        if (session.isLegacyTimestamp()) {
+            return new SqlTime(block.getLong(position, 0), session.getTimeZoneKey());
+        }
+        else {
+            return new SqlTime(block.getLong(position, 0));
```

Comment:
IMO, the "Introduce new TIME/TIMESTAMP semantics SPI types" commit should be squashed with "Introduce new TIME/TIMESTAMP semantics to SQL types" commit.